compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java

```diff
@@ -32,10 +33,15 @@
 
     private BallerinaUserHome(Environment environment, Path ballerinaUserHomeDirPath) {
         this.ballerinaUserHomeDirPath = ballerinaUserHomeDirPath;
+        Path settingsTomlPath = Paths.get(String.valueOf(ballerinaUserHomeDirPath),
+                ProjectConstants.SETTINGS_FILE_NAME);
         Path remotePackageRepositoryPath = ballerinaUserHomeDirPath.resolve(ProjectConstants.REPOSITORIES_DIR)
                 .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
         try {
             Files.createDirectories(remotePackageRepositoryPath);
+        } catch (AccessDeniedException ae) {
```

Comment:
We are not trying to create the Settings.toml here. We are only creating the repo cache.